# Введение в Data Science и машинное обучение
# Задачи

## 1. Введение

### 1.4. Pandas, Dataframes, нет, панды тут ни при чем

А теперь используем эти методы на знаменитом [титаническом датасэте](https://stepik.org/media/attachments/course/4852/titanic.csv)! 

Загрузите датасэт, посмотрите на датафрэйм и ответьте на вопросы

Число колонок в представленном датафрэйме `_`, а число строк `_`. Тип float имеют `_` колонки, int - `_`, и object `_`

In [1]:
import pandas as pd

In [2]:
titanic = pd.read_csv('data\\titanic.csv')

In [3]:
titanic.shape

(891, 12)

In [4]:
titanic.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

In [5]:
titanic.dtypes.value_counts()

object     5
int64      5
float64    2
dtype: int64

Число колонок в представленном датафрэйме `12`, а число строк `891`. Тип float имеют `2` колонки, int - `5`, и object `5`

### 1.5. Фильтрация данных

`?`

У какой доли студентов из [датасэта](https://stepik.org/media/attachments/course/4852/StudentsPerformance.csv) в колонке `lunch` указано `free/reduced`?

In [6]:
students_performance = pd.read_csv('data\\StudentsPerformance.csv')

In [7]:
students_performance.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2  female        group B             master's degree      standard   
3    male        group A          associate's degree  free/reduced   
4    male        group C                some college      standard   

  test preparation course  math score  reading score  writing score  
0                    none          72             72             74  
1               completed          69             90             88  
2                    none          90             95             93  
3                    none          47             57             44  
4                    none          76             78             75

In [8]:
students_performance.columns

Index(['gender', 'race/ethnicity', 'parental level of education', 'lunch',
       'test preparation course', 'math score', 'reading score',
       'writing score'],
      dtype='object')

In [9]:
len(students_performance[students_performance['lunch'] == 'free/reduced']) \
    / len(students_performance)

0.355